firmware/screen/tools/long_cmd_fix.py

```
@@ -0,0 +1,27 @@
+#
+# CC/CXX/AR/LINK absolute-path replacement and TEMPFILE wrapping live in the
+# POST-phase script (long_cmd_fix_post.py), because the platform script that
+# runs AFTER this pre-phase will overwrite whatever Replace() we do here.
+Import("env")
+import os
+
+pio_home = os.environ.get('PLATFORMIO_CORE_DIR',
+    os.path.join(os.environ.get('USERPROFILE',''), '.platformio'))
+tc_bin = os.path.join(pio_home, 'packages', 'toolchain-xtensa-esp32s3', 'bin')
+
+env.PrependENVPath("PATH", tc_bin)
```

Ruff (0.15.10)

[error] Undefined name `env`

(F821)